hashring/hashring.go

```diff
@@ -215,53 +215,43 @@ func (r *HashRing) ServerCount() int {
 	return count
 }
 
+// Lookup returns the owner of the given key.
 func (r *HashRing) Lookup(key string) (string, bool) {
-	r.servers.RLock()
-
-	iter := r.servers.tree.IterAt(int(r.hashfunc([]byte(key))))
-	if iter.Nil() {
-		r.servers.RUnlock()
+	strs := r.LookupN(key, 1)
```

Comment:
Sorry, I was just saying that I like what you've done, I think it is an improvement.